moshi/src/main/java/com/squareup/moshi/JsonAdapter.kt

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi
+
+import com.squareup.moshi.internal.NonNullJsonAdapter
+import com.squareup.moshi.internal.NullSafeJsonAdapter
+import okio.Buffer
+import okio.BufferedSink
+import okio.BufferedSource
+import okio.IOException
+import org.intellij.lang.annotations.Language
```

Comment:
Oh interesting, which artifact does this come from, kotlin-stdlib?

Reply:
yup, JetBrains annotations includes it